pkg/egressgateway/policy.go

```diff
@@ -299,3 +286,94 @@ func ParsePolicyConfigID(cenp *v2alpha1.CiliumEgressNATPolicy) types.NamespacedN
 		Name: cenp.Name,
 	}
 }
+
+// ParseCEGP takes a CiliumEgressGatewayPolicy CR and converts to PolicyConfig,
+// the internal representation of the egress gateway policy
+func ParseCEGP(cegp *v2.CiliumEgressGatewayPolicy) (*PolicyConfig, error) {
```

Comment:
nit: maybe rename also `ParsePolicy` to `ParseCENP`? (Or even better refactor the 2 into a single function is possible that takes the API version as parameter)

Reply:
Rename sounds good! Will have another look at consolidating the two functions, but maybe rather as a follow-up.